mitodl/open-discussions
# add basic integration of CKEditor w/ Easy Image support

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #1456 
closes #1455 
closes #1326 

#### What's this PR do?

This adds a basic integration for CKEditor, including:

- implementing a new `ArticleEditor` component which does the editing and display of saved articles
- an `ARTICLE_UI` feature flag
- adding options and support for editing an article to the post create page
- logic to display an article with the read-only version of the editor on the post detail page
- validation for the post create page
- a backend API for authenticating with CKEditor's easy image service

I think that's about it.

#### How should this be manually tested?

Basically, I think the thing to do is get everything set up and then play around with it, make sure you can save articles with images, view them, etc. To get set up you'll need to get the secrets for CKEditor and add them to `.env`, alongside the `ARTICLE_UI` feature flag.

If you do that you should a new option in the post create page which will allow you to 'write an article'. You should also be able to add images to the article and save it. On the post detail page the article you wrote should be display by the read-only mode of the article editor.

#### Screenshots (if appropriate)

![addarticle](https://user-images.githubusercontent.com/6207644/49464881-1b672880-f7ca-11e8-937f-64ab9ffddff4.png)
![addarticleasdfasdfasdfffffval](https://user-images.githubusercontent.com/6207644/49464889-21f5a000-f7ca-11e8-84ed-ca266c0c0163.png)
![addarticleasdfasdfasdf](https://user-images.githubusercontent.com/6207644/49464893-25892700-f7ca-11e8-87f3-47ae3e29d948.png)
![addarticleasdfasdfasdfffff](https://user-images.githubusercontent.com/6207644/49464899-291cae00-f7ca-11e8-8489-330c660f617b.png)
